create.ps1

```diff
@@ -1,200 +1,220 @@
+        Write-Verbose "Module [$ModuleName] is already imported."
+    }
+    else {
+        # If module is not imported, but available on disk then import
+        if (Get-Module -ListAvailable -Verbose:$false | Where-Object { $_.Name -eq $ModuleName }) {
+            $module = Import-Module $ModuleName -Verbose:$false
+            Write-Verbose "Imported module [$ModuleName]"
+        }
+        else {
+            # If the module is not imported, not available and not in the online gallery then abort
+            throw "Module [$ModuleName] is not available. Please install the module using: Install-Module -Name [$ModuleName] -Force"
+        }
+    }
+    #>
 
+    # Check if we should try to correlate the account
+    if ($actionContext.CorrelationConfiguration.Enabled) {
+        $correlationField = $actionContext.CorrelationConfiguration.accountField
+        $correlationValue = $actionContext.CorrelationConfiguration.accountFieldValue
 
-    #Check correlation before create
-    if ($correlationCheckResult.externalId -eq $correlationValue) {
-        $aRef = ($correlationCheckResult.$correlationField)
+        if ($null -eq $correlationField) {
+            Write-Warning "Correlation is enabled but not configured correctly."
+        }
 
-        $query = "UPDATE persons 
-                    SET email = @email
-                    ,achternaam = @achternaam
-                    ,voornaam = @voornaam
-                    ,tussenvoegsel = @tussenvoegsel
-                    ,gebruikersnaam = @gebruikersnaam
-                    ,geslacht = @geslacht
-                    ,createtime = datetime()
-                    WHERE externalId = @externalId"
+        # Write logic here that checks if the account can be correlated in the target system
+        # Requesting authorization token
 
-        $sqlParameters = $account.psobject.properties | ForEach-Object -begin { $h = @{} } -process { $h."$($_.Name)" = $_.Value } -end { $h }
+        try {
+            $query = "SELECT $correlationField,achternaam,gebruikersnaam FROM persons WHERE $correlationField = '$correlationValue'"
+            $correlationCheckResult = Invoke-SqliteQuery -Query $query -DataSource $database -Verbose:$verbose
+            if ($correlationCheckResult.externalId -eq $correlationValue) {
+                $correlatedAccount = $true
+            }
+        }
+        catch {
+            write-error "$($_)"
+            $ex = $PSItem
+            
+            $auditMessage = "Error querying data from SQL Lite DB. Error: $($ex.Exception.Message)"
+            Write-Verbose "Error at Line '$($ex.InvocationInfo.ScriptLineNumber)': $($ex.InvocationInfo.Line). Error: $($ex.Exception.Message)"
+            
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    Action  = "CreateAccount" # Optionally specify a different action for this audit log
+                    Message = $auditMessage
+                    IsError = $false
+                })
```

Comment:
Is the auditlog necessary? Maybe just write an auditlog in case the action itself (create) fails.